tests/core/test_mnemonic.py

```diff
@@ -0,0 +1,522 @@
+# Originally from: https://github.com/trezor/python-mnemonic
+#
+# Copyright (c) 2013 Pavol Rusnak
+# Copyright (c) 2017 mruddy
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy of
+# this software and associated documentation files (the "Software"), to deal in
+# the Software without restriction, including without limitation the rights to
+# use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies
+# of the Software, and to permit persons to whom the Software is furnished to do
+# so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+# WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+# CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+#
+import pytest
+
+from eth_account.hdaccount.mnemonic import (
+    Mnemonic,
+    ValidationError,
+    normalize_string,
+)
+
+
+def test_failed_checksum():
+    mnemo = Mnemonic("english")
+    assert not mnemo.check(
+        "bless cloud wheel regular tiny venue bird web grief security dignity zoo"
+    )
+
+
+@pytest.mark.parametrize("language,word", [
+    ("english", "security"),
+    ("french", "baguette"),
+    ("spanish", "leche"),
+    ("italian", "cordata"),
+    ("czech", "nitro"),
+    ("japanese", "あらためる"),
+    ("korean", "강북"),
+    ("chinese_simplified", "也"),
+    ("chinese_traditional", "滅"),
+])
+def test_detection(language, word):
+    assert language == Mnemonic.detect_language(word)
+
+
+def test_undetected_language():
+    with pytest.raises(ValidationError):
+        Mnemonic.detect_language("xxxxxxx")
+
+
+def test_expand_word():
+    m = Mnemonic("english")
+    assert "" == m.expand_word("")
+    assert " " == m.expand_word(" ")
+    assert "access" == m.expand_word("access")  # word in list
+    assert "access" == m.expand_word("acce")  # unique prefix expanded to word in list
```

Comment:
Can this test get converted into a parametrized one.

Reply:
I think I'll keep this as is (because it checks a lower-level API), and make the other one parameterized

Reply:
I believe that parametrization is the right thing for this because all of these checks are checking the same thing, rather than multiple checks that build on different types of validation.  Thus, if the case `assert "" == m.expand_word("")` is broken **and** `assert " " == m.expand_word(" ")` is also broken, it will only expose the first issue, hiding the second issue since the test won't get to that assertion.  That means that in the case that there is a bug in multiple of these cases, you will only ever be able to see the first one that fails.